From 3d5c0f46f10bcb26f054af9ab2cf1d910148f9d5 Mon Sep 17 00:00:00 2001
From: Petri Lehtinen <petri@digip.org>
Date: Mon, 6 Sep 2010 20:46:14 +0300
Subject: [PATCH] Run Sphinx without the -W flag when building documentation

In "make html", don't use the -W flag with Sphinx. This makes it
possible to create the documentation with Sphinx 1.0 without errors,
as the warning about using old-style C markup isn't turned to an
error.

Don't build documentation in "make check". Instead, add a new make
target "check-doc" to build the documentation with the -W flag.
---
 Makefile.am     |    3 ++-
 doc/Makefile.am |    2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index a141cd5..173c7de 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,7 +1,8 @@
 EXTRA_DIST = CHANGES LICENSE README.rst
 SUBDIRS = doc src test
 
-check-local: html
+check-doc:
+	$(MAKE) SPHINXOPTS_EXTRA=-W html
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = jansson.pc
diff --git a/doc/Makefile.am b/doc/Makefile.am
index 74a715c..1f2d855 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -3,7 +3,7 @@ EXTRA_DIST = conf.py apiref.rst changes.rst conformance.rst		\
 	ext/refcounting.py
 
 SPHINXBUILD = sphinx-build
-SPHINXOPTS = -d _build/doctrees -W
+SPHINXOPTS = -d _build/doctrees $(SPHINXOPTS_EXTRA)
 
 html-local:
 	$(SPHINXBUILD) -b html $(SPHINXOPTS) $(srcdir) _build/html
-- 
1.7.3.5

